dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java

```diff
@@ -1035,43 +1074,180 @@ public void Test_File_Comparison_Is_Case_Sensitive()
+        api.createAppDescriptor(file, admin);
+
+        final String appKey = dataGen.getKey();
+        //generate secrets
+        AppSecrets secrets = builder1.withKey(appKey)
+                .withHiddenSecret("p1", "secret-1")
+                .withHiddenSecret("p2", "secret-2")
                 .build();
         //Save it
         api.saveSecrets(secrets, site, admin);
         final Optional<AppSecrets> secretsOptional = api.getSecrets(appKey, site, admin);
-        final AppSecrets appSecretsPostSave = secretsOptional.get();
+        Assert.assertTrue(secretsOptional.isPresent());
+        secrets = secretsOptional.get();
 
-        //AES only supports key sizes of 16, 24 or 32 bytes.
+        //AES only supports security Key of sizes of 16, 24 or 32 bytes.
         final String password = RandomStringUtils.randomAlphanumeric(32);
-        final Key key = AppsUtil.generateKey(password);
-        final Path exportSecretsFile = api.exportSecrets(key, true, null, admin);
+        final Key securityKey = AppsUtil.generateKey(password);
+
+        //Now that we have a valid key lets dump our selection of secrets
+        final Map<String,Set<String>> appKeysBySite = ImmutableMap.of(site.getIdentifier(), ImmutableSet.of(appKey));
+        final Path exportSecretsFile = api.exportSecrets(securityKey, false, appKeysBySite, admin);
         assertTrue(exportSecretsFile.toFile().exists());
 
-        //Remove so we can re import them.
+        //Remove the secret we dumped we can re import it.
         api.deleteSecrets(appKey, site, admin);
 
-        final Map<String, List<AppSecrets>> secretAppsBySiteId = api.importSecrets(exportSecretsFile, key, admin);
-        assertFalse(secretAppsBySiteId.isEmpty());
-        assertTrue(secretAppsBySiteId.containsKey(site.getIdentifier()));
-        final List<AppSecrets> appSecretsBySite = secretAppsBySiteId.get(site.getIdentifier());
-        assertFalse(appSecretsBySite.isEmpty());
-        final AppSecrets importedSecrets = appSecretsBySite.get(0);
-        assertEquals(importedSecrets, appSecretsPostSave);
+        //import it
+        api.importSecretsAndSave(exportSecretsFile, securityKey, admin);
 
-        api.saveSecrets(importedSecrets, site, admin);
+        //verify
         final Optional<AppSecrets> secretsOptionalPostImport = api.getSecrets(appKey, site, admin);
         assertTrue(secretsOptionalPostImport.isPresent());
+        final AppSecrets restoredSecrets = secretsOptionalPostImport.get();
+
+        assertEquals(restoredSecrets.getKey(),secrets.getKey());
+        assertEquals(restoredSecrets.getSecrets().size(),secrets.getSecrets().size());
+        for (final Entry<String, Secret> entry : secrets.getSecrets().entrySet()) {
+            final Secret originalSecret = entry.getValue();
+            final Secret restoredSecret = restoredSecrets.getSecrets().get(entry.getKey());
+            assertTrue(originalSecret.equals(restoredSecret));
```

Comment:
Codacy found an issue: [Use assertEquals(x, y) instead of assertTrue(x.equals(y))](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6286472)